libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -0,0 +1,214 @@
+import { CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+    AfterContentInit,
+    AfterViewChecked,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { fromEvent, Subject, Subscription } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { RtlService } from '../utils/services/rtl.service';
+
+const CARD_MINIMUM_WIDTH = 320; // in px; 20rem max card size
+const CARD_GAP_WIDTH = 16; // gap=1rem==16px
+
+@Directive({ selector: '[fdCardDef]' })
```

Comment:
What is this directive needed for?

Reply:
This is structural directive. this is getting used to achieve "z" flow structure from initial column layout structure.
By query getting all these directive attached element and re-arranging based on "z" flow.